File: examples/performance/primes.py
SIZE = 1_000_000
primes = [2, 3]

n = 5
i = 2
j = 1
limit = pow(n, 0.5)

while (i < SIZE):
    isPrime = True

    j = 1
    while isPrime and primes[j] < limit and j < i:
        if n % primes[j] == 0:
            isPrime = False
        j += 1

    if isPrime:
        primes.append(n)
        i += 1

        print(n)

    n += 2
    limit = pow(n, 0.5)

File: examples/performance/Primes.java
class Primes {
    private static final int SIZE = 1_000_000;
    private static final int[] primes = new int[SIZE];

    public static void main(String[] args) {
        primes[0] = 2;
        primes[1] = 3;

        var n = 5;
        var i = 2;
        var j = 1;
        var limit = Math.pow(n, 0.5);

        while (i < SIZE)
        {
            var isPrime = true;

            for (j = 1; isPrime && primes[j] < limit && j < i; j++)
            {
                if (n % primes[j] == 0)
                {
                    isPrime = false;
                }
            }

            if (isPrime)
            {
                primes[i] = n;
                i++;

                System.out.println(n);
            }

            n += 2;
            limit = Math.pow(n, 0.5);
        }
    }
}
